Dockerfile

```diff
@@ -0,0 +1,86 @@
+FROM ubuntu:16.04
+
+# Surpress Upstart errors/warning
+RUN dpkg-divert --local --rename --add /sbin/initctl
+RUN ln -sf /bin/true /sbin/initctl
+
+# Let the conatiner know that there is no tty
+ENV DEBIAN_FRONTEND noninteractive
+
+# Update base image
+# Add sources for latest nginx
+# Install software requirements
+RUN apt-get update && \
+apt-get install -y software-properties-common && \
+nginx=stable && \
+add-apt-repository ppa:nginx/$nginx && \
+apt-get update && \
+apt-get upgrade -y && \
+BUILD_PACKAGES="supervisor nginx php7.0-fpm git php7.0-mysql php7.0-curl php7.0-gd php7.0-intl php7.0-mcrypt php7.0-sqlite php7.0-tidy php7.0-xmlrpc php7.0-xsl php7.0-pgsql php7.0-ldap pwgen unzip php7.0-zip curl php-mbstring" && \
+apt-get -y install $BUILD_PACKAGES && \
+curl -sS https://getcomposer.org/installer -o composer-setup.php && \
+php composer-setup.php --install-dir=/usr/local/bin --filename=composer && \
+apt-get remove --purge -y software-properties-common && \
+apt-get autoremove -y && \
+apt-get clean && \
+apt-get autoclean && \
+echo -n > /var/lib/apt/extended_states && \
+rm -rf /var/lib/apt/lists/* && \
+rm -rf /usr/share/man/?? && \
+rm -rf /usr/share/man/??_*
+
+# tweak nginx config
+RUN sed -i -e"s/worker_processes  1/worker_processes 5/" /etc/nginx/nginx.conf && \
+sed -i -e"s/keepalive_timeout\s*65/keepalive_timeout 2/" /etc/nginx/nginx.conf && \
+sed -i -e"s/keepalive_timeout 2/keepalive_timeout 2;\n\tclient_max_body_size 100m/" /etc/nginx/nginx.conf && \
+echo "daemon off;" >> /etc/nginx/nginx.conf
+
+# tweak php-fpm config
+RUN sed -i -e "s/;cgi.fix_pathinfo=1/cgi.fix_pathinfo=0/g" /etc/php/7.0/fpm/php.ini && \
```

Comment:
it is better to have this in a custom .ini file like z-pagevamp.ini and copy it here than do all this magic

Reply:
Ya. That would be more readable but benefit of this way: change only what you need to and works with almost any versions rather than replacing whole config file :)